sql/sql_alloc.h

```diff
@@ -44,15 +44,6 @@ class Sql_alloc
   static void operator delete[](void *ptr, MEM_ROOT *mem_root)
   { /* never called */ }
   static void operator delete[](void *ptr, size_t size) { TRASH(ptr, size); }
-#ifdef HAVE_purify
```

Comment:
I believe this fragment was originally meant to overcome some Valgrind limitation (not being able to handle empty base classes properly, I guess). What was the warning here?

Reply:
Currently I can't reproduce a warning. 
I may removed it because it was removed in 5.7 by upstream:
https://github.com/mysql/mysql-server/commit/955e335b5729a7d74e88f00e6bd46b88a0682670

Reply:
Changes reverted.